### DISCUSSION TITLE
SUN2000 8/12KTL doesn't have 43xxxx range registers

### DISCUSSION THREAD

**Discussion by hjalmarnilsson** [wlcrs/huawei_solar#351] [General]
Hi,
I have a 2019 sun2000 8/12 KTL, ie not the M0 version (mine is grey, not the newer white wo lcd)
It seems impossible to get the firmware that matches a 8/12 KTL off the https://support.huawei.com/ website, and I'm a bit dubious to download a fw for a later sun2000 12ktl V100R001 variant..
When reading the register definitions for the 8/12KTL there are no registers in the 43000 range so it can't read the timezone register that it seems to try to read when setting it up.

I get the following error message, that makes me belive that I have a proper connection on the usb-rs485 and it's wired up correctly.

Any pointers? is it possible to configure the integration to not use "newer" registers when it tests to communicate with it via any config file in config/custom_components/huawei_solar folder?
I get the below error when trying to add the integration
"/usr/local/lib/python3.10/site-packages/huawei_solar/huawei_solar.py", line 308, in _do_read
    raise ReadException(
huawei_solar.exceptions.ReadException: Got error while reading from register 43006 with length 1: Exception Response(131, 3, IllegalAddress)

**Comment by wlcrs** [comment]
No, there is nothing to configure in this integration: the underlying library tries to detect everything automatically. I don't want to expose any configuration parameters to the end-user as they increase the support load (people creating tickets because they don't understand what the parameters mean, etc).

I do not have access to an inverter of your generation, nor to the documentation of it. I invite you to contribute the necessary patches to https://gitlab.com/Emilv2/huawei-solar/ so that I can properly detect and work with your system. You can have a look at how I detect optimizer / power meter / battery presence in `huawei_solar.py` to get started.
